applications/crossbar/src/modules/cb_rates.erl

```diff
@@ -252,10 +268,47 @@ on_successful_validation(Id, Context) ->
 %% resource.
 %% @end
 %%--------------------------------------------------------------------
--spec summary(cb_context:context()) -> cb_context:context().
-summary(Context) ->
-    crossbar_doc:load_view(?CB_LIST, [], Context, fun normalize_view_results/2).
-
+-spec summary(cb_context:context(), ne_binary(), ne_binary(), ne_binaries()) -> cb_context:context().
+summary(Context, _AccountId, _AuthId, []) ->
+    lager:debug("loading global rates"),
+    summary(Context, [?WH_RATES_DB]);
+summary(Context, AccountId, AccountId, Parents) ->
+    lager:debug("loading self rates"),
+    DBs = [?WH_RATES_DB
+           | lists:map(
+               fun account_rates_db/1
+               ,Parents ++ [AccountId]
+              )],
+    summary(Context, DBs);
+summary(Context, AccountId, _AuthId, _Parents) ->
+    lager:debug("loading client rates"),
+    summary(Context, [account_rates_db(AccountId)]).
+
+-spec summary(cb_context:context(), ne_binaries()) -> cb_context:context().
+summary(Context, DBs) ->
+    Context1 = crossbar_doc:load_view(?CB_LIST, [{'databases', DBs}, 'include_docs'], Context, fun normalize_view_results/2),
+    Context2 = case cb_context:resp_status(Context1) of
+                   'fatal' ->
+                       cb_context:set_resp_status(cb_context:set_resp_data(Context1, []), 'success');
+                   _ ->
+                       Context1
+               end,
+    Data = lists:foldl(fun merge_acc/2
+                      ,[]
+                      ,cb_context:resp_data(Context2)),
+    Data1 = [Val || {_K, Val} <- Data],
+    cb_context:set_resp_data(Context2, Data1).
+
+-spec merge_acc({ne_binary(), ne_binary()}, list()) -> list().
```

Comment:
This spec still seems wrong. `Val` can't be only ne_binary() if you're testing it for 'undefined' below. And `Acc` is what? Seems like `view_results()` is the right type?